### PR TITLE
[PORT] Port customer_pricing_in_product_view to V10

### PR DESCRIPTION
**Pull request by gdgellatly**
Is a cleaner PR from #245 rebased onto 10.0. Just the customer context files.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Now that I re-visit this module, I think that a better name is _product_customer_pricing_, as the main area is product, and this reflects in less words the feature. What do you think?

**Comment by gdgellatly** [comment]
@pedrobaeza Lol that was pretty much the original name. I had to change to
get module accepted in first place. I hate the current name so happy to
change.

**Comment by pedrobaeza** [comment]
Really? And it was me who told you about the change? I'm crazy... :stuck_out_tongue_closed_eyes:

**Comment by gdgellatly** [comment]
@pedrobaeza I don't remember now, was years ago.  Anyhow I changed the name to customer_pricing_in_product_view which pretty well explains the module and roadmap.

File: customer_pricing_in_product_view/README.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+Configuration
+=============
+
+In order for the module to be useful you will need to have enabled both
+variants and advanced pricelists under the Sales / Settings menu.
+
+Usage
+=====
+
+Navigate to a customer (form view)
```
Comment (pedrobaeza) [inline customer_pricing_in_product_view/README.rst]: Put a numbered list with `#.`

File: customer_pricing_in_product_view/README.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+In the resulting tree view an extra column called price will show the customer price.
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/135/10.0
+
+Known issues / Roadmap
+======================
+
+Previous versions of this module allowed the customers name to be typed directly in the view.
```
Comment (pedrobaeza) [inline customer_pricing_in_product_view/README.rst]: Put this into a bullet list prepending points with `*`

File: customer_pricing_in_product_view/README.rst [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+While fields.dummy has been deprecated the functionality can still be acheived by an unstored many2one
```
Comment (pedrobaeza) [inline customer_pricing_in_product_view/README.rst]: s/acheived/achieved

File: customer_pricing_in_product_view/README.rst [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+on product.template in the same way as pricelist_id is currently stored.
+
+However, the way in which the context is set presents two issues.
+
+* The customers name rather than id is stored in context when referencing self.
+* The tree view does not dynamically change context in version 10 from search bar entries.
+
+(In fact it makes the pricelist functionality of the tree view in core Odoo irrelevant)
+
+If these issues are resolved and there is a way to get
```
Comment (pedrobaeza) [inline customer_pricing_in_product_view/README.rst]: Is this half redacted?

File: customer_pricing_in_product_view/__manifest__.py [hunk 5]
```diff
@@ -21,18 +21,13 @@
 
 {
     'name': 'Display Customer Price in Product View',
-    'version': '1.1.1',
+    'version': '10.0.1.0.0',
     'category': 'Sales',
-    'author': "O4SB - Graeme Gellatly,Odoo Community Association (OCA)",
-    'website': 'http://www.o4sb.com',
+    'author': "Odoo Community Association (OCA)",
```
Comment (pedrobaeza) [inline customer_pricing_in_product_view/__manifest__.py]: You should include yourself in the author list

File: customer_pricing_in_product_view/models/res_partner.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+##############################################################################
```
Comment (pedrobaeza) [inline customer_pricing_in_product_view/models/res_partner.py]: Use short headers like the one in https://github.com/OCA/maintainer-tools/blob/master/template/module/__init__.py

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by gdgellatly** [comment]
@pedrobaeza changes as requested

**Review by lasley — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**